samples/ClassroomLabs/Modules/Library/Tools/Readme.md

```diff
@@ -1,20 +1,138 @@
 1. Get a local copy of the example [hogwarts.csv](https://raw.githubusercontent.com/Azure/azure-devtestlab/master/samples/ClassroomLabs/Modules/Library/Tools/hogwarts.csv) file and example [charms.csv](https://raw.githubusercontent.com/Azure/azure-devtestlab/master/samples/ClassroomLabs/Modules/Library/Tools/charms.csv) file
 1. Launch a PowerShell session
 1. Ensure [Azure PowerShell](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps) installed
-1. Update the example CSV files to configure the resources to be created.  For addiitonal labs, create additional lines in the CSV file.  The CSV files can be modified diretly with Microsoft Excel.
-1. Execute the Bulk Lab Creation script with the following commandline:
+1. Update the example CSV files to configure the resources to be created.  For additional labs, create additional lines in the CSV file.  The CSV files can be modified directly with Microsoft Excel.
+1. Import the functions in the script by typing:
+```powershell
+Import-Module LabCreationLibrary.psm1
+```
+
+## Examples
+The functions are generic and can be composed together to achieve different aims. In the following examples, we load the configuration information from a CSV file. The examples work the same if the information is loaded from a database. You need to substitute the first function with a database retrieving one.
+
+The full code for the example is immediately after the title in parenthesis.
+
+### Publish all the labs in a CSV file ([Examples/PublishAll.ps1](./Examples/PublishAll.ps1))
+
 ```powershell
-.\LabCreator.ps1 -CsvConfigFile hogwarts.csv
+".\hogwarts.csv" | Import-LabsCsv | Publish-Labs
 ```
+* `Import-LabsCsv` loads the configuration information from the csv file. It also loads schedule information for each lab from a separate file.
+* `Publish-Labs` publishes the labs and it is the natural end to all our pipelines. You can specify how many concurrent threads to use with the parameter `ThrottleLimit`.
+
+### Publish all the labs in a JSON file ([Examples/PublishAllJson.ps1](./Examples/PublishAllJson.ps1))
+You can store your configuration information in JSON form as well (i.e., file, Azure CosmoDb, other).
+
+```powershell
+Get-Content -Raw -Path "hogwarts.json" | ConvertFrom-Json | Publish-Labs
+```
+
+### Publish all the labs with a particular tag ([Examples/PublishByTag.ps1](./Examples/PublishByTag.ps1))
+
+```powershell
+".\hogwarts.csv"  | Import-LabsCsv | Select-Lab -SomeTags Maths, Science | Publish-Labs
+```
+
+In the example csv configuration file, the `Tags` column contains the emails of the professors that are giving a certain course. You can create the labs for a particular professor with:
+
+```powershell
+".\hogwarts.csv"  | Import-LabsCsv | Select-Lab -SomeTags bool1@hotmail.com | Publish-Labs
```

Comment:
We have a "LabOwner" field in the CSV - can use that to enable this scenario

Reply:
Not exactly as you can have multiple professors for a course. But I agree, the all tags situation is a bit weird.